sql/api/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -4061,7 +4061,14 @@ object functions {
    * @group string_funcs
    * @since 2.3.0
    */
-  def ltrim(e: Column, trimString: String): Column = Column.fn("ltrim", lit(trimString), e)
+  def ltrim(e: Column, trimString: String): Column = ltrim(e, lit(trimString))
+
+  /**
+   * Trim the specified character string from left end for the specified string column.
+   * @group string_funcs
+   * @since 4.0.0
+   */
+  def ltrim(e: Column, trim: Column): Column = Column.fn("ltrim", trim, e)
```

Comment:
Is `trim` the expression for the trimString? Looks a bit weird to call this parameter `trim`. It's also not consistent with the function doc of expression `StringTrim`, which names this parameter as `trimStr`

Reply:
unfortanately, the argument names were already not consistent between python and scala, in many functions.

in this PR I use `trim` to be consistent with python side `btrim`

Reply:
https://github.com/apache/spark/blob/78135dc204e4fc18829a3698f1f253f90b7da301/python/pyspark/sql/functions/builtin.py#L13399